lib/core/stores/local.ts

```diff
@@ -207,6 +210,228 @@ export class LocalPromiseStore implements IPromiseStore {
     tags: Record<string, string> | undefined,
     limit: number | undefined,
   ): AsyncGenerator<DurablePromise[], void> {
-    return this.storage.search(id, state, tags, limit);
+    return this.storage.search(id, state, tags, limit) as AsyncGenerator<DurablePromise[], void>;
+  }
+}
+
+export class LocalScheduleStore implements IScheduleStore {
+  private promiseStorage: IPromiseStorage;
+  private storage: IScheduleStorage;
+
+  constructor(storage: MemoryStorage) {
+    this.promiseStorage = storage.promises;
+    this.storage = storage.schedules;
+
+    this.startControlLoop();
```

Comment:
Let's move the control loop logic down to the `LocalStore` instead. That way promises and schedules will be nicely  isolated.